icefoganalytics/travel-authorization
# Dev Tooling: Finally Enable Eslint for Front-End

# Context

Set up Eslint in the front-end. This helps ensure consistency across the codebase.
It also make development easier, since it highlights certain kinds of errors that are hard to find normally.

# Implementation
Note that I'm hoisting the .gitignore and .prettier.yaml so they get loaded as well. They are one level higher to match the external setup, so the paths make sense when read.

# Testing Instructions

1. Lint a file like `web/src/modules/travel-authorizations/pages/travel-form-edit/details-tab/DetailsFormCard.vue` via 
    ```sh
    dev web npm run lint src/modules/travel-authorizations/pages/travel-form-edit/details-tab/DetailsFormCard.vue
    ````
    > You should get an error around prop ordering.
